### PR TITLE
[xpu] support: each parameter has different lr in merged_momentum

### PR DESCRIPTION
**Pull request by zhouquan32**
### PR Category
Custom Device

### PR Types
Bug fixes

### Description
The current version of XPU requires all parameters to use the same learning rate. However, GUP supports scenarios where each parameter can have a different learning rate for merged momentum. To address this discrepancy, a temporary modification is necessary.

### CODE REVIEW

**Review by tizhou86 — DISMISSED** [review]
LGTM

File: paddle/phi/kernels/xpu/merged_momentum_kernel.cc [hunk 1]
```diff
@@ -83,8 +84,10 @@ void MergedMomentumKernel(
   std::vector<XPUType*> grad_list(op_num);
   std::vector<XPUType*> velocity_out_list(op_num);
   std::vector<XPUType*> param_out_list(op_num);
+  std::vector<const float*> lr_list(op_num);
   std::vector<int> sizes(op_num);
   std::vector<float> l2_weight_decay(op_num);
+  int lr_len = learning_rate.size();
```
Comment (QingshuChen) [inline paddle/phi/kernels/xpu/merged_momentum_kernel.cc]: 这边有要求lr_len == op_num 或者lr_len = 1 吗?  其他情况是否要拋异常?

**Comment by houj04** [comment]
修改了对学习率的处理方式之后，对应的单测是不是也要改？之前是多个tensor都用同一个学习率，现在改成每个tensor用自己的学习率，新的这种情况，在单测里面覆盖到了吗？

**Review by dynamicheart — APPROVED** [review]
LGTM

**Review by QingshuChen — APPROVED** [review]
